.github/workflows/integration_test.yml

```diff
@@ -0,0 +1,40 @@
+name: "Integration test"
+# NOTE: We're just running the tests that require earthdata login here; we
+#       don't distinguish yet between unit and integration tests yet.
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+      - "development"
+
+
+jobs:
+  test:
+    name: "Test"
+    if: "github.repository == 'icesat2py/icepyx'"
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          fetch-depth: 0
+
+      - uses: "./.github/actions/install-icepyx"
+        with:
+          python-version: "3.12"
+
+      - name: "Run tests"
+        env:
+          EARTHDATA_USERNAME: "${{ secrets.EARTHDATA_USERNAME }}"
+          EARTHDATA_PASSWORD: "${{ secrets.EARTHDATA_PASSWORD }}"
```

Comment:
@JessicaS11 these are the secrets we need added to this repository :point_up:

Reply:
I added EARTHDATA_PASSWORD and CODECOV_TOKEN to the org/repo secrets. EARTHDATA_USERNAME is included as part of a session fixture within `test_behind_NSIDC_API_login.py`.